src/main/java/org/elasticsearch/common/io/PathUtils.java

```diff
@@ -75,7 +75,22 @@ public static Path get(URI uri) {
             return Paths.get(uri);
         }
     }
-    
+
+    /**
+     * Tries to resolve the given path against the list of available roots.
+     *
+     * If path starts with one of the listed roots, it returned back by this method, otherwise null is returned.
+     */
+    public static Path get(Path[] roots, String path) {
+        Path normalizedPath = DEFAULT.getPath(path).normalize();
+        for (Path root : roots) {
+            if(normalizedPath.startsWith(root)) {
+                return normalizedPath;
+            }
+        }
```

Comment:
Can we avoid DEFAULT here? This code assumes that every Path in `roots` uses that filesystem. Technically, they could be different. I would prefer if we avoid get()-like methods for cases like this, and just resolved simply it, like this:

    for (Path root : roots) {
      Path normalizedPath = root.resolve(path).normalize();
      if (normalizedPath.startsWith(root)) {
        ...